byterun/major_gc.c

```diff
@@ -872,9 +872,14 @@ static void cycle_all_domains_callback(struct domain* domain, void* unused,
   CAMLassert(atomic_load(&num_domains_to_sweep) == 0);
   CAMLassert(atomic_load(&num_domains_to_ephe_sweep) == 0);
 
-  caml_stw_empty_minor_heap(domain, (void*)0, participating);
+  barrier_status b = caml_global_barrier_begin();
```

Comment:
Might be better to contain this in the minor_gc.c functions.
Maybe we have a `caml_stw_empty_minor_heap_from_stw` that packages all this in one place?

Reply:
Yea, separating those two out is a good idea.